### DISCUSSION TITLE
Auto starting/stopping machine

### DISCUSSION THREAD

**Discussion by Pusnow** [containers/podman#13923] [Ideas]
When I do `podman run` (or other commands) without starting podman machine, podman CLI only returns error messages.

```
$ podman run -it --rm debian:bullseye                           
Cannot connect to Podman. Please verify your connection to the Linux system using `podman system connection list`, or try `podman machine init` and `podman machine start` to manage a new Linux VM
Error: unable to connect to Podman. failed to create sshClient: Connection to bastion host (ssh://core@localhost:54774/run/user/501/podman/podman.sock) failed.: dial tcp [::1]:54774: connect: connection refused
```

Instead, I think it would be better if machine starts automatically and runs the command (auto starting).

Also, auto stoping could be useful too.

Automatically stop the machine when no containers are running for a certain amount of time.

Any thought?

**Comment by rhatdan** [comment]
Autostopping, I could see, autostarting might be more difficult, since we would need mechanisms in Windows and MAC to understand to start the machine.

I guess we could instrument Podman to attempt to start a machine if no socket existed or if the socket connection was rejected.  I think this would need to be controlled in containers.conf.

WDYT @n1hility @vrothberg @jwhonce @baude @ashley-cui

**Reply by vrothberg** [reply]
I like the idea of auto-starting.  What does Docker do?

**Comment by n1hility** [comment]
Docker desktop auto-starts on boot with a launch item managed by a tray app. We could add a command to install an on boot start (which some of the tray work we have could use as well) One thing is if we did do on demand podman run auto-start it would only cover the podman cli. If anyone uses the docker style api clients they would need to manually start those.

**Reply by rhatdan** [reply]
I am fine if this only worked with Podman CLI.
